packages/serverpod/lib/src/database/adapters/postgres/database_connection.dart

```diff
@@ -377,10 +395,18 @@ class DatabaseConnection {
           postgresTransaction?.executionContext ?? _postgresConnection;
 
       var result = await context.execute(
-        query,
+        switch (parameters) {
+          (Map<String, Object?> _) => pg.Sql.named(query),
+          (QueryParametersNamed _) => pg.Sql.named(query),
+          _ => query,
+        },
         timeout: timeout,
         ignoreRows: ignoreRows,
         queryMode: simpleQueryMode ? pg.QueryMode.simple : null,
+        parameters: switch (parameters) {
+          (QueryParameters _) => parameters.parameters,
+          _ => parameters,
+        },
```

Comment:
Suggestion: If we only allow `parameters` to be of type `QueryParameters` we should be able to simply call `paramters?.parameters`.

Then we can remove this redundant switch statement.